poly/deal.go

```diff
@@ -276,68 +276,74 @@ func (p *Promise) ConstructPromise(secretPair *config.KeyPair,
 	return p
 }
 
-/* Initializes a Promise for unmarshalling
+/* Initializes a Deal for unmarshalling
  *
  * Arguments
  *    t           = the minimum number of shares needed to reconstruct the secret
  *    r           = the minimum number of positive Response's needed to cerifty the
- *                  promise
+ *                  deal
  *    n           = the total number of insurers.
- *    suite       = the suite used within the Promise
+ *    suite       = the suite used within the Deal
  *
  * Returns
- *   An initialized Promise ready to be unmarshalled
+ *   An initialized Deal ready to be unmarshalled
  */
-func (p *Promise) UnmarshalInit(t, r, n int, suite abstract.Suite) *Promise {
```

Comment:
I think this and other methods called UnmarshalInit should just be called Init.  That's the standard name we've been using to initialize otherwise-"empty" objects; doesn't matter if they subsequently get used for unmarshmaling or something else.